### DISCUSSION TITLE
Sudden Build Failures and Unsuccessful Re-deployments After Vercel CLI Update (34.1.7)

### DISCUSSION THREAD

**Discussion by lgauthier1** [vercel/community#6714] [Bug Report]
### Page to Investigate

NA

### Steps to Reproduce

Hello,

I'm encountering a sudden issue where my project no longer builds successfully, despite there being no changes to the code or dependencies. I've confirmed that dependencies are consistent using npm ci, which indicates that no unexpected changes have occurred. The issue coincided with an update to the Vercel CLI, which seems to be the only significant change.

Error Details:

```js
Build Completed in /vercel/output [31s]
Deploying outputs...
    at qze (/var/task/sandbox.js:405:12920)
    at w8r (/var/task/sandbox.js:407:33497)
    at async Mze (/var/task/sandbox.js:407:33236)
    at async P8r (/var/task/sandbox.js:409:4108)
    at async Qze (/var/task/sandbox.js:409:4952)
    at async N8r (/var/task/sandbox.js:409:1388)
    at async Promise.all (index 2)
    at async D8r (/var/task/sandbox.js:409:479)
    at async zze (/var/task/sandbox.js:407:36120)
    at async f_r (/var/task/sandbox.js:425:2334) {
  idx: 7
}

Injected preview comments middleware, you can disable this in your project settings.

Deployment completed
--
21:13:52.667 | Uploading build cache [239.21 MB]...
21:13:56.683 | Build cache uploaded: 4.016s

```

Moreover, when I attempt to re-deploy older builds that were previously successful and fully operational, they now fail, indicating that the issue may stem from a change at Vercel's end rather than any alterations in my project's setup.

It is puzzling that the build process ultimately reports as successful ("Build Completed") despite the errors listed. I am looking for insights or suggestions on how to resolve these errors or any further diagnostic steps I should take. Any help would be greatly appreciated!

Thank you!

**Comment by CamaradePopoff** [comment]
Same issue here. Current vercel CLI version is 34.1.7 according to the build logs.
Using the VERCEL_CLI_VERSION environment variable I tried to force the CLI version back to an older stack version, without success.

Yet in the past this kind of forced downgrade worked and helped me avoid a bug of the latest vercel CLI, but not this time.

I tried:
VERCEL_CLI_VERSION="vercel@34.1.1"
VERCEL_CLI_VERSION="vercel@33.7.1"

The build logs show that the version I want is well taken into account and installed prior to the build itself, but in the end it still displays the same strange "/var/task/sandbox" error block, and the deployed NodeJS app does not get started.

**Comment by techieinprague** [comment]
I'm facing the same issue. The build fails before it even starts. Did you find any fixes?

@CamaradePopoff @lgauthier1 is the only solution to wait it out until Vercel fixes it from their side?

**Reply by lgauthier1** [reply]
Hey @techieinprague, not yet. 

We're trying it on another simple project (Node 20 / simple backend project node + Apollo), and it's working with Vercel@34.1.1. 

We're facing an issue with Node 18, so we'll try upgrading to Node 20 to see if it resolves the problem. (as our simple project)

But any advice from the Vercel team (@amyegan 👋 )  would be great to help us understand this stack trace, and how to fix it

**Reply by techieinprague** [reply]
I found out the core issue. Apparently I hit the free-tier build limit. If you're using the Hobby tier there's a [limit of how many builds](https://vercel.com/docs/limits/overview) you can run in 24 hrs.

**Reply by lgauthier1** [reply]
Thank you for your feedback, @techieinprague , but we are not on the free tier and in my usage dashboard i don't see any out of limit KPI.

**Reply by amyegan** [reply]
@lgauthier1 I hope you already found a solution. If you're still experiencing build errors, please [start a new discussion](https://github.com/vercel/community/discussions/new) and include the error log, an outline of what you have tried, and a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that I can use to help with debugging.

**Reply by lgauthier1** [reply]
Let me know if you need a new discussion, if not please find below the minimal reproductible example:
https://github.com/lgauthier1/vercel-build-issue 

```bash
[00:39:19.561] Running build in Washington, D.C., USA (East) – iad1
[00:39:19.739] Retrieving list of deployment files...
[00:39:20.209] Downloading 6 deployment files...
[00:39:20.678] Restored build cache
[00:39:20.803] Running "vercel build"
[00:39:21.635] Vercel CLI 34.1.10
[00:39:22.800] Installing dependencies...
[00:39:29.012] npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
[00:39:29.125] npm WARN deprecated debug@4.1.1: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
[00:39:30.906] 
[00:39:30.907] added 246 packages in 8s
[00:39:30.907] 
[00:39:30.907] 28 packages are looking for funding
[00:39:30.907]   run `npm fund` for details
[00:39:31.519] Build Completed in /vercel/output [9s]
[00:39:31.660] Deploying outputs...
[00:39:31.677]     at qze (/var/task/sandbox.js:427:12920)
[00:39:31.677]     at x8r (/var/task/sandbox.js:429:33497)
[00:39:31.677]     at async Mze (/var/task/sandbox.js:429:33236)
[00:39:31.677]     at async L8r (/var/task/sandbox.js:431:4108)
[00:39:31.677]     at async Qze (/var/task/sandbox.js:431:4952)
[00:39:31.677]     at async k8r (/var/task/sandbox.js:431:1388)
[00:39:31.677]     at async Promise.all (index 4)
[00:39:31.677]     at async R8r (/var/task/sandbox.js:431:479)
[00:39:31.677]     at async zze (/var/task/sandbox.js:429:36120)
[00:39:31.677]     at async h_r (/var/task/sandbox.js:447:2334) {
[00:39:31.677]   idx: 7
[00:39:31.677] }
[00:39:39.466] 
[00:39:39.710] Deployment completed
[00:39:41.868] Uploading build cache [13.28 MB]...
[00:39:42.229] Build cache uploaded: 360.668ms
```

**Comment by junyakawakami-shells** [comment]
We are facing a similar problem...

``` bash

Build Completed in /vercel/output [1m]
--
14:16:26.412 | Deploying outputs...
14:16:26.461 | at qze (/var/task/sandbox.js:405:12920)
14:16:26.462 | at w8r (/var/task/sandbox.js:407:33497)
14:16:26.462 | at async Mze (/var/task/sandbox.js:407:33236)
14:16:26.462 | at async P8r (/var/task/sandbox.js:409:4108)
14:16:26.462 | at async Qze (/var/task/sandbox.js:409:4952)
14:16:26.462 | at async N8r (/var/task/sandbox.js:409:1388)
14:16:26.462 | at async Promise.all (index 172)
14:16:26.462 | at async D8r (/var/task/sandbox.js:409:479)
14:16:26.462 | at async zze (/var/task/sandbox.js:407:36120)
14:16:26.462 | at async f_r (/var/task/sandbox.js:425:2334) {
14:16:26.462 | idx: 2511
14:16:26.462 | }

~~~~

An unexpected error happened when running this build. We have been notified of the problem. This may be a transient error. If the problem persists, please contact Vercel Support https://vercel.com/help

```

Redeployment of a previously successful deployment fails.
Specifying the same VERCEL_CLI_VERSION as a previously successful deployment did not work.

We are currently contacting the Vercel development team to see if they can help. If we make any progress, I will share it.

If anyone has any solutions, please let me know🙏

**Comment by lgauthier1** [comment]
After analysis, we have confirmed that the error is present. However, it does not prevent the build from completing and does not affect the functionality of our API.

This log has been observed in multiple projects and does not seem to cause any blockage. Therefore, I am closing this discussion.

It would be great if we could get some answers from the Vercel team regarding the root cause of this issue.

**Reply by amyegan** [reply]
I'm happy to hear that your work is not blocked by this. There are many possible causes for build errors, so I can't determine the actual cause without digging into it further. If you're able to share a small example repo, I can take a closer look.

**Reply by lgauthier1** [reply]
Thanks for your answer !

You'll find below a very simple [repo](https://github.com/lgauthier1/vercel-build-issue 
), ready to deploy (follow the README), and you should see the same error as described below:
<img width="558" alt="image" src="https://github.com/vercel/community/assets/37870500/193860e2-6c6f-4092-a618-59cb59de8d2a">

Keep in mind that despite this strange error log, the build is OK and the API is working well: https://vercel-build-issue.vercel.app/api

https://github.com/lgauthier1/vercel-build-issue

**Reply by aldosch** [reply]
Hi @lgauthier1 we have rolled out a fix to all build containers ~10 hours ago. Could you please perform a new deployment to test if these errors are still appearing? 

These logs were not real build errors. They were internal warnings from our build container logic which should have been omitted. Only the first line was omitted which meant that multi-line warnings were partially visible. Engineering confirmed that the errors can be safely ignored as functionality was not affected. 

Please let me or Amy know if the errors are still appearing or if you have any other questions. Thanks for bringing this to our attention.

**Reply by aldosch** [reply]
hey @techieinprague and @junyakawakami-shells could you please retest too? Thanks! 🙏

**Reply by lgauthier1** [reply]
Hi @aldosch,

Thank you for confirming my analysis. Indeed, these logs (in red) looked like errors but did not affect the deployed build.
Following your correction, I confirm that after a new deployment, the logs no longer contain this strange stack trace.
Many thanks to you for the support. 😉

Conversations could be resolved !

```bash 
Running build in Washington, D.C., USA (East) – iad1
--
09:23:00.288 | Retrieving list of deployment files...
09:23:00.781 | Downloading 6 deployment files...
09:23:02.038 | Restored build cache
09:23:02.167 | Running "vercel build"
09:23:03.204 | Vercel CLI 34.1.10
09:23:05.569 | Installing dependencies...
09:23:07.422 |  
09:23:07.422 | up to date in 1s
09:23:07.422 |  
09:23:07.422 | 28 packages are looking for funding
09:23:07.422 | run `npm fund` for details
09:23:08.642 | Build Completed in /vercel/output [3s]
09:23:08.931 | Deploying outputs...
09:23:09.689 | Injected preview comments middleware, you can disable this in your project settings.
09:23:16.746 |  
09:23:16.997 | Deployment completed
09:23:19.303 | Uploading build cache [13.28 MB]...
09:23:19.627 | Build cache uploaded: 323.275ms
```
